upup/pkg/fi/cloudup/awstasks/launchconfiguration.go

```diff
@@ -172,14 +183,19 @@ func (e *LaunchConfiguration) Find(c *fi.Context) (*LaunchConfiguration, error)
 	actual.SecurityGroups = securityGroups
 
 	// Find the root volume
-	for _, b := range lc.BlockDeviceMappings {
-		if b.Ebs == nil || b.Ebs.SnapshotId != nil {
-			// Not the root
+	for i, b := range lc.BlockDeviceMappings {
+		if b.Ebs == nil {
 			continue
 		}
-		actual.RootVolumeSize = b.Ebs.VolumeSize
-		actual.RootVolumeType = b.Ebs.VolumeType
-		actual.RootVolumeIops = b.Ebs.Iops
+		switch i {
+		case 0:
```

Comment:
I think if we create it, it will be :-)

The "root device" is actually defined by the image, and the "magic" is https://github.com/kubernetes/kops/blob/af4df08b694e2a1f8814a7b3649060477be67c86/upup/pkg/fi/cloudup/awstasks/launchconfiguration.go#L242

You could also figure it out by matching the device, but I think that we can always create with root at pos 0, and if someone created one with a different order that we would just force a spurious update / rolling update.

Of course I'm not sure if AWS ever reorders the BlockDeviceMappings in the slice.  I doubt they do, but they might.